File: ECnotes/Sem2/LivExamples/VB/Ch22WebParts/App_Code/HalloweenCounter.vb
Imports Microsoft.VisualBasic

Namespace murach

    Public Class HalloweenCounter
        Inherits WebPart

        Public Sub New()
            Me.Title = "Days to Halloween"
        End Sub

        Protected Overrides Sub RenderContents(ByVal writer As HtmlTextWriter)
            Dim dtmToday As Date = Date.Today
            Dim dtmHalloween As Date = New Date(Date.Today.Year, 10, 31)

            If dtmToday > dtmHalloween Then
                dtmHalloween = dtmHalloween.AddYears(1)
            End If

            Dim tsSpan As TimeSpan = dtmHalloween - dtmToday
            Dim sContent As String = "There are " & tsSpan.Days & " shopping days until Halloween."

            writer.Write(sContent)
        End Sub

    End Class

End Namespace

File: ECnotes/Sem2/LivExamples/CS/Ch22WebParts/App_Code/HalloweenCounter.cs
using System;
using System.Web.UI;
using System.Web.UI.WebControls.WebParts;

namespace murach
{
	public class HalloweenCounter : WebPart
	{
		public HalloweenCounter()
		{
			this.Title = "Days to Halloween";
		}

		protected override void RenderContents(HtmlTextWriter writer)
		{
			DateTime today = DateTime.Today;

			DateTime halloween = new DateTime(DateTime.Today.Year, 10, 31);

			if (today > halloween)
				halloween = halloween.AddYears(1);

			TimeSpan span = halloween - today;

			string content = "There are " + span.Days + " shopping days until Halloween.";
             
			writer.Write(content);
		}
	}
}
